### PR TITLE
Do not merge! Updates to build 6079043

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt [hunk 1]
```diff
@@ -73,7 +73,7 @@ fun JetnewsApp() {
 @Composable
 private fun AppContent(openDrawer: () -> Unit) {
     Crossfade(JetnewsStatus.currentScreen) { screen ->
-        Surface(color = (+MaterialTheme.colors()).background) {
+        Surface(color = (MaterialTheme.colors()).background) {
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt]:
```suggestion
        Surface(color = MaterialTheme.colors().background) {
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt [hunk 2]
```diff
@@ -73,7 +72,7 @@ fun ArticleScreen(postId: String) {
             title = {
                 Text(
                     text = "Published in: ${post.publication?.name}",
-                    style = (+MaterialTheme.typography()).subtitle2
+                    style = (MaterialTheme.typography()).subtitle2
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt]:
```suggestion
                    style = MaterialTheme.typography().subtitle2
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt [hunk 3]
```diff
@@ -136,7 +135,7 @@ private fun FunctionalityNotAvailablePopup(onDismiss: () -> Unit) {
         text = {
             Text(
                 text = "Functionality not available \uD83D\uDE48",
-                style = (+MaterialTheme.typography()).body2
+                style = (MaterialTheme.typography()).body2
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt]:
```suggestion
                style = MaterialTheme.typography().body2
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt [hunk 4]
```diff
@@ -66,57 +68,69 @@ private val codeBlockBackground = Color(0xfff1f1f1.toInt())
...
-            Text(
-                text = metadata.author.name,
-                style = typography.caption.withOpacity(0.87f)
-            )
-            Text(
-                text = "${metadata.date} • ${metadata.readTimeMinutes} min read",
-                style = typography.caption.withOpacity(0.6f)
-            )
+            Spacer(LayoutHeight(4.dp))
+            ProvideEmphasis(emphasis = EmphasisLevels().high) {
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/article/PostContent.kt]: Much nicer!

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt [hunk 5]
```diff
@@ -71,11 +72,13 @@ fun HomeScreen(openDrawer: () -> Unit) {
...
-    Text(
-        modifier = Spacing(top = 16.dp, left = 16.dp, right = 16.dp),
-        text = "Top stories for you",
-        style = ((+MaterialTheme.typography()).subtitle1).withOpacity(0.87f)
-    )
+    ProvideEmphasis(emphasis = EmphasisLevels().high) {
+        Text(
+            modifier = LayoutPadding(left = 16.dp, top = 16.dp, right = 16.dp, bottom = 0.dp),
+            text = "Top stories for you",
+            style = (MaterialTheme.typography()).subtitle1
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt]:
```suggestion
            style = MaterialTheme.typography().subtitle1
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt [hunk 6]
```diff
@@ -96,15 +99,17 @@ private fun HomeScreenSimpleSection(posts: List<Post>) {
...
-    Text(
-        modifier = Spacing(16.dp),
-        text = "Popular on Jetnews",
-        style = ((+MaterialTheme.typography()).subtitle1).withOpacity(0.87f)
-    )
+    ProvideEmphasis(emphasis = EmphasisLevels().high) {
+        Text(
+            modifier = LayoutPadding(16.dp),
+            text = "Popular on Jetnews",
+            style = ((MaterialTheme.typography()).subtitle1)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt]:
```suggestion
            style = MaterialTheme.typography().subtitle1
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt [hunk 7]
```diff
@@ -47,28 +47,36 @@ fun PostCardPopular(post: Post) {
...
-                                        "${post.metadata.readTimeMinutes} min read",
-                                style = ((+MaterialTheme.typography()).body2).withOpacity(0.6f)
-                            )
+                        Column(modifier = LayoutPadding(16.dp)) {
+
+                            val emphasisLevels = EmphasisLevels()
+                            ProvideEmphasis(emphasis = emphasisLevels.high) {
+                                Text(
+                                    text = post.title,
+                                    style = (MaterialTheme.typography()).h6,
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt]:
```suggestion
                                    style = MaterialTheme.typography().h6,
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt [hunk 8]
```diff
@@ -47,28 +47,36 @@ fun PostCardPopular(post: Post) {
...
+                                    maxLines = 2,
+                                    overflow = TextOverflow.Ellipsis
+                                )
+                            }
+                            ProvideEmphasis(emphasis = emphasisLevels.high) {
+                                Text(
+                                    text = post.metadata.author.name,
+                                    style = ((MaterialTheme.typography()).body2)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt]:
```suggestion
                                    style =(MaterialTheme.typography().body2
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt [hunk 9]
```diff
@@ -47,28 +47,36 @@ fun PostCardPopular(post: Post) {
...
+                                )
+                            }
+                            ProvideEmphasis(emphasis = emphasisLevels.high) {
+                                Text(
+                                    text = "${post.metadata.date} - " +
+                                            "${post.metadata.readTimeMinutes} min read",
+                                    style = ((MaterialTheme.typography()).body2)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt]:
```suggestion
                                    style = MaterialTheme.typography().body2
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt [hunk 10]
```diff
@@ -45,27 +45,32 @@ import com.example.jetnews.ui.navigateTo
...
 fun AuthorAndReadTime(post: Post) {
     Row {
-        val textStyle = ((+MaterialTheme.typography()).body2).withOpacity(0.6f)
-        Text(text = post.metadata.author.name, style = textStyle)
-        Text(
-            text = " - ${post.metadata.readTimeMinutes} min read",
-            style = textStyle
-        )
+
+        val textStyle = ((MaterialTheme.typography()).body2)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt]:
```suggestion
        val textStyle = MaterialTheme.typography().body2
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt [hunk 11]
```diff
@@ -74,9 +79,12 @@ fun PostCardSimple(post: Post) {
         Clickable(onClick = {
             navigateTo(Screen.Article(post.id))
         }) {
-            Row(modifier = Spacing(16.dp)) {
-                PostImage(modifier = Spacing(right = 16.dp), post = post)
-                Column(modifier = Flexible(1f)) {
+            Row(modifier = LayoutPadding(16.dp)) {
+                PostImage(
+                    modifier = LayoutPadding(0.dp, 0.dp, right = 16.dp, bottom = 0.dp),
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt]: Use named params for all.

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt [hunk 12]
```diff
@@ -95,18 +103,23 @@ fun PostCardHistory(post: Post) {
...
-            Row(modifier = Spacing(all = 16.dp)) {
-                PostImage(modifier = Spacing(right = 16.dp), post = post)
-                Column(modifier = Flexible(1f)) {
-                    Text(
-                        text = "BASED ON YOUR HISTORY",
-                        style = ((+MaterialTheme.typography()).overline).withOpacity(0.38f)
-                    )
+            Row(modifier = LayoutPadding(all = 16.dp)) {
+                PostImage(
+                    modifier = LayoutPadding(0.dp, 0.dp, right = 16.dp, bottom = 0.dp),
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt]: Use named params for all.

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt [hunk 13]
```diff
@@ -95,18 +103,23 @@ fun PostCardHistory(post: Post) {
...
+                    post = post
+                )
+                Column(modifier = LayoutFlexible(1f)) {
+                    ProvideEmphasis(emphasis = EmphasisLevels().medium) {
+                        Text(
+                            text = "BASED ON YOUR HISTORY",
+                            style = ((MaterialTheme.typography()).overline)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt]:
```suggestion
                            style = MaterialTheme.typography().overline
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt [hunk 14]
```diff
@@ -95,18 +103,23 @@ fun PostCardHistory(post: Post) {
...
+                        )
+                    }
                     PostTitle(post = post)
                     AuthorAndReadTime(post)
                 }
                 VectorImage(
-                    modifier = Spacing(top = 8.dp, bottom = 8.dp),
+                    modifier = LayoutPadding(0.dp, top = 8.dp, right = 0.dp, bottom = 8.dp),
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCards.kt]: Use named params for all.

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt [hunk 15]
```diff
@@ -133,8 +133,8 @@ private fun TabWithSections(
             sections.forEach { (section, topics) ->
                 Text(
                     text = section,
-                    modifier = Spacing(16.dp),
-                    style = (+MaterialTheme.typography()).subtitle1)
+                    modifier = LayoutPadding(16.dp),
+                    style = (MaterialTheme.typography()).subtitle1)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt]:
```suggestion
                    style = MaterialTheme.typography().subtitle1
                )
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt [hunk 16]
```diff
@@ -62,21 +61,21 @@ fun SelectTopicButton(
 private fun DrawSelectTopicButtonOn() {
     DrawShape(
         shape = CircleShape,
-        color = (+MaterialTheme.colors()).primary
+        color = (MaterialTheme.colors()).primary
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt]:
```suggestion
        color = MaterialTheme.colors().primary
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt [hunk 17]
```diff
@@ -62,21 +61,21 @@ fun SelectTopicButton(
...
     )
-    DrawVector(+vectorResource(R.drawable.ic_check))
+    DrawVector(vectorResource(R.drawable.ic_check))
 }
 
 @Composable
 private fun DrawSelectTopicButtonOff() {
-    val borderColor = ((+MaterialTheme.colors()).onSurface).copy(alpha = 0.12f)
+    val borderColor = ((MaterialTheme.colors()).onSurface).copy(alpha = 0.12f)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt]:
```suggestion
    val borderColor = MaterialTheme.colors().onSurface.copy(alpha = 0.12f)
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt [hunk 18]
```diff
@@ -62,21 +61,21 @@ fun SelectTopicButton(
...
     DrawBorder(
         shape = CircleShape,
         border = Border(borderColor, 2.dp)
     )
     DrawVector(
-        vectorImage = +vectorResource(R.drawable.ic_add),
-        tintColor = (+MaterialTheme.colors()).primary
+        vectorImage = vectorResource(R.drawable.ic_add),
+        tintColor = (MaterialTheme.colors()).primary
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt]:
```suggestion
        tintColor = MaterialTheme.colors().primary
```

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt (hunk 15 above)
Comment (JoseAlcerreca) [reply JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt]: You're missing a (

**Comment by JoseAlcerreca** [comment]
Changes merged into the `develop` branch. Thanks!

**Pull request closed without merging**